code/Controllers/CMSMain.php

```diff
@@ -744,8 +746,36 @@ public function getEditForm($id = null, $fields = null)
             $fields->push($liveLinkField = new HiddenField("AbsoluteLink", false, $record->AbsoluteLink()));
             $fields->push($liveLinkField = new HiddenField("LiveLink"));
             $fields->push($stageLinkField = new HiddenField("StageLink"));
+            $fields->push($archiveWarningMsg = new HiddenField("ArchiveWarningMessage"));
             $fields->push(new HiddenField("TreeTitle", false, $record->TreeTitle));
 
+            // Get all page's descendants
+            $record->collateDescendants(true, $descendants);
+            if (!$descendants) {
+                $descendants = [];
+            }
+
+            // Get all campaigns that the page and its descendants belong to
+            $inChangeSetIDs = ChangeSetItem::get_for_object($record)->column('ChangeSetID');
+            foreach ($descendants as $page) {
+                $inChangeSetIDs = array_merge($inChangeSetIDs, ChangeSetItem::get_for_object($page)->column('ChangeSetID'));
+            }
+            $inChangeSets = ChangeSet::get()->filter(['ID' => $inChangeSetIDs, 'State' => ChangeSet::STATE_OPEN]);
+
+            if (count($descendants) > 0 && $inChangeSets->count() > 0) {
+                $archiveWarningTerm = 'CMSMain.ArchiveWarningWithChildrenAndCampaigns';
+            } else if (count($descendants) > 0) {
+                $archiveWarningTerm = 'CMSMain.ArchiveWarningWithChildren';
+            } else if ($inChangeSets->count() > 0) {
+                $archiveWarningTerm = 'CMSMain.ArchiveWarningWithCampaigns';
+            } else {
+                $archiveWarningTerm = 'CMSMain.ArchiveWarning';
+            }
+
+            $numCampaigns = ChangeSet::singleton()->i18n_pluralise($inChangeSets->count());
+            $numCampaigns = mb_strtolower($numCampaigns);
+            $archiveWarningMsg->setValue(_t($archiveWarningTerm, ' ', [ 'NumCampaigns' => $numCampaigns ]));
```

Comment:
Please set proper translation fallbacks so we can see immediately what the translation is meant to be (and proper fallback if translation files were removed for some reason in the future).